quiz.js

```diff
@@ -0,0 +1,459 @@
+
+        // add a random animation class to each question
+        numRandom = Math.floor(Math.random() * animations.length);
+        output += animations[numRandom];
+
+        output += "'>";
+        output += "<legend class='question-heading'>" + questions[x].number + ". " + questions[x].title + "</legend>";
+
+        for (var j = 0; j < questions[x].answers.length; j++) {
+            output += "<div class='questions'>";
+            output += "<label><input type='radio' name='q" + questions[x].number + "' onclick='processChecked(this)'>";
+            output += questions[x].answers[j] + "</label>";
+            output += "</div>";
+        }
+
+        output += "</div>";
+    }
+
+    document.getElementById("questions-container").innerHTML = output;
+
+    // add an active class to the first question to show it
+    document.querySelector(".inner").classList.add("active");
+
+});
+
+
+
+/* ========
+
+Start Quiz
+
+=========== */
+
+function startQuiz() {
+    document.getElementById("instructions-container").style.display = 'none';
+    document.getElementById("one").classList.remove("no-display");
+}
+
+/* show next button */
+function showNextBtn() {
+    document.getElementById("next").classList.remove("no-display");
+}
+
+/* make next button active */
+function activateNextBtn() {
+    document.getElementById("next").classList.remove("inactive");
+    document.getElementById("next").removeAttribute("disabled");
+    document.getElementById("next").classList.add("animate-right-arrow");
+}
+
+/* make next button inactive */
+function inactivateNextBtn() {
+    document.getElementById("next").classList.add("inactive");
+    document.getElementById("next").setAttribute("disabled", "");
+    document.getElementById("next").classList.remove("animate-right-arrow");
+}
+
+
+/* ========
+
+Check if question has been answered
+
+=========== */
+
+/* store the questions that have been answered */
+var answeredQuestions = [];
+
+/* everytime a radio is selected, check to see if the current question has been answered, if not, add it to the answeredQuestions array */
+function processChecked(el) {
+
+    // if the current radio box is checked
+    if (el.checked === true) {
+
+        // if the question has not been answered yet (is not in the answered questions array)
+        if (answeredQuestions.indexOf(currentQuestion) == -1) {
+
+            // add to the answered questions array
+            answeredQuestions.push(currentQuestion);
+
+            updateProgress(currentQuestion * 10);
+
+            // when the last question is answered, enabled and show the calc score button
+            if (currentQuestion == 10) {
+                document.getElementById("calculate").classList.remove("inactive");
+                document.getElementById("calculate").removeAttribute("disabled");
+            }
+
+            activateNextBtn();
+
+        }
+
+    }
+
+}
+
+
+/* Update progress bar */
+
+function updateProgress(progress) {

```

Comment:
https://codepen.io/tomtomau/pen/MvJjXd

You can do progress bars with mostly CSS as a way to reduce the amount you do in JS, also for shits and giggles.